File: RB101_Programming_Foundations/Small Probems/Easy8/double_char.rb
def repeater(string)
  double_string = ''

  string.each_char do |char|
    double_string += char * 2
  end
  double_string
end

p repeater('Hello') == "HHeelllloo"
p repeater("Good job!") == "GGoooodd  jjoobb!!"
p repeater('') == ''

File: RB101_Programming_Foundations/Small Probems/Easy8/Python/double_char.py
# Write a method that takes a string, and returns a new string in which every character is doubled.


'''
Input: string

Output: new string with every character doubled

Rules


Data structure / Algorithms
    create an array of size len(string)
    iterate through each character in string
        add to the array the current string * 2


'''


def repeater(string):
    array = [""] * len(string)

    for idx, char in enumerate(string):
        array[idx] = char * 2

    return ''.join(array)

print(repeater('Hello') == "HHeelllloo")
print(repeater("Good job!") == "GGoooodd  jjoobb!!")
print(repeater('') == '')